src/api/updateFiles.ts

```diff
@@ -401,3 +417,139 @@ const checkResultsForErrors = (results: DocumentBulkResponse[]): void => {
     }
   })
 }
+
+const checkChangeSetDocumentsConsistency = (appState: AppState) => async (
+  repoId: string,
+  changeSet: ChangeSet
+): Promise<boolean> => {
+  const paths = Object.keys(changeSet)
+  const repoDocumentKey = `${repoId}:/`
+
+  // Set of all document keys for every document affected in the changeset
+  const documentKeysSet = paths.reduce<Set<string>>((documentKeys, path) => {
+    // Add document key for full path
+    documentKeys.add(`${repoId}:${path}`)
+
+    // Add document keys for each parent path
+    const parentPaths = getParentPathsOfPath(path)
+    for (const parentPath of parentPaths) {
+      documentKeys.add(`${repoId}:${parentPath}`)
+    }
+
+    // Return the set accumulator
+    return documentKeys
+  }, new Set([repoDocumentKey]))
+
+  // Array of the document keys
+  const documentKeys = [...documentKeysSet]
+
+  // A graph of each parent document key to a set of child document keys
+  const documentKeyGraph = documentKeys.reduce<Map<string, string[]>>(
```

Comment:
Write a comment giving an example of the final data structure that is documentKeyGraph. Something like:

```
/*
const documentKeyGraph = {
  "repoId:/dir1": [
    "repoId:/dir1/file1.txt",
    "repoId:/dir1/file2.txt",
    "repoId:/dir1/subdir"
  ],
  "repoId:/dir1/subdir": [
    "repoId:/dir1/subdir/somefile.tar"
  ]
}
*/
```